daemon/config/config.go

```diff
@@ -171,10 +171,9 @@ type CommonConfig struct {
 	// Proxies holds the proxies that are configured for the daemon.
 	Proxies `json:"proxies"`
 
-	// TrustKeyPath is used to generate the daemon ID and for signing schema 1 manifests
-	// when pushing to a registry which does not support schema 2. This field is marked as
-	// deprecated because schema 1 manifests are deprecated in favor of schema 2 and the
-	// daemon ID will use a dedicated identifier not shared with exported signatures.
+	// TrustKeyPath is deprecated. The daemon ID is now written to the "engine_id"
+	// file within the daemon's data-root.
+	// Deprecated: this configuration option is no longer used.
```

Comment:
Can we just remove this field?

Reply:
Let me see if we can remove (it's been a while since I looked closely at the changes, only just did a rebase); perhaps this was still to produce a more friendly error? (need to check)